math/src/field/smallprimefield/mod.rs

```diff
@@ -0,0 +1,493 @@
+use crate::fft;
+
+use super::{
+    traits::{FieldElement, StarkField},
+    QuadExtensionA,
+};
+use core::{
+    convert::{TryFrom, TryInto},
+    fmt::{Debug, Display, Formatter},
+    mem,
+    ops::{Add, AddAssign, Div, DivAssign, Mul, MulAssign, Neg, Sub, SubAssign},
+    slice,
+};
+use utils::{
+    collections::Vec,
+    string::{String, ToString},
+    AsBytes, ByteReader, ByteWriter, Deserializable, DeserializationError, Randomizable,
+    Serializable,
+};
+
+#[cfg(test)]
+mod tests;
+
+// Number of bytes needed to represent field element
+const ELEMENT_BYTES: usize = mem::size_of::<u64>();
+
+// SMALL PRIME FIELD ELEMENT
+// ================================================================================================
+
+/// Represents a base field element with a small modulus.
+/// Primarily for testing, so we can hand calculate expected outputs.
+/// Note that all the operations here are naive field operations without optimizations.
+/// We also don't end up needing to use the twoadic roots anymore,
+/// since we can factor MODULUS - 1 easily. Meaning, any factor root of MODULUS - 1 can be found.
+/// Internal values are stored in their canonical form in the range [0, MODULUS). The backing type is
+/// `u64`.
```

Comment:
Looks great! A couple of minor comments:

1. Would be great to explain the meaning of generic parameters here. For example, initially, I thought that `G` was the 2-addic root of unity, but it seems like it is actually the primitive element in the field.
2. We should add a blank line in the comments between the first line and the rest. This way, the comment will be formatted better in the docs (you can run `cargo doc --open` to generate and view documentation).
3. Nit: we should get rid of the blank line between the comment and the struct.